cli/src/args.rs

```diff
@@ -104,20 +105,30 @@ impl RgbArgs {
         }
 
         let provider = FsBinStore::new(stock_path.clone())?;
-        let mut stock = Stock::load(provider, true).map_err(WalletError::WalletPersist).or_else(|err| {
-            if matches!(err, WalletError::Deserialize(DeserializeError::Decode(DecodeError::Io(ref err))) if err.kind() == ErrorKind::NotFound) {
-                if self.verbose > 1 {
-                    eprint!("stock file is absent, creating a new one ... ");
+        let mut stock = Stock::load(provider, true)
+            .map_err(WalletError::WalletPersist)
+            .or_else(|err| {
+                if let WalletError::WalletPersist(PersistenceError(ref err)) = err {
+                    if let Some(DeserializeError::Decode(DecodeError::Io(io_err))) =
+                        err.downcast_ref::<DeserializeError>()
+                    {
+                        if io_err.kind() == ErrorKind::NotFound {
```

Comment:
I do not think we need that deep construction; we can do the same way we did before with errors in errors

```suggestion
                let Some(DeserializeError::Decode(DecodeError::Io(io_err))) = err.0.downcast_ref::<DeserializeError>() else {
                    eprintln!("stock file is damaged, failing");
                    return Err(err)
                };
                if io_err.kind() == ErrorKind::NotFound {
```